convolve_fft test fails on some platforms

### Description

On Debian, we currently rebuilt all Python packages to include Python 3.10 as supported version. With both Python versions (3.9.9, 3.10), I now see a failure on some platforms in the following doctest example:
https://github.com/astropy/astropy/blob/bf18eb5f1a5cf3b2835c67d3fa303d29b587aa66/astropy/convolution/convolve.py#L622-L623

### Expected behavior

All tests should pass

### Actual behavior

```
_____________ [doctest] astropy.convolution.convolve.convolve_fft ______________
329 
330     Examples
331     --------
332     >>> convolve_fft([1, 0, 3], [1, 1, 1])
333     array([0.33333333, 1.33333333, 1.        ])
334 
335     >>> convolve_fft([1, np.nan, 3], [1, 1, 1])
336     array([0.5, 2. , 1.5])
337 
338     >>> convolve_fft([1, 0, 3], [0, 1, 0])
Expected:
    array([ 1.00000000e+00, -3.70074342e-17,  3.00000000e+00])
Got:
    array([1.00000000e+00, 3.70074342e-17, 3.00000000e+00])

/<<PKGBUILDDIR>>/.pybuild/cpython3_3.9/build/astropy/convolution/convolve.py:338: DocTestFailure
=========================== short test summary info ============================
FAILED astropy/convolution/convolve.py::astropy.convolution.convolve.convolve_fft
```
[Full build log (i386) here](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=i386&ver=4.3.1-4%2Bb1&stamp=1637567550&raw=0)

### Steps to Reproduce

Build and test astropy with  python 3.9.9 or 3.10.0 

### System Details
```
Platform: Linux-5.10.0-9-amd64-i686-with-glibc2.32

Executable: /usr/bin/python3.9

Full Python Version: 
3.9.9 (main, Nov 16 2021, 10:24:31) 
[GCC 11.2.0]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.21.4
Scipy: 1.7.1
Matplotlib: 3.5.0
h5py: 3.5.0
Pandas: not available
PyERFA: 2.0.0.1
Cython: 0.29.24
Scikit-image: not available
asdf: not available

Using Astropy options: remote_data: none.
```
Same happens for python 3.10.0+, and for the arm64, s390x, ppc64el, and ppc64 platforms.
I didn't check this for Astropy 5.0RC2 yet, but since the example is unchanged, I expect the same failure.

--- a/astropy/convolution/convolve.py
+++ b/astropy/convolution/convolve.py
@@ -619,7 +619,7 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0.,
     >>> convolve_fft([1, np.nan, 3], [1, 1, 1])
     array([0.5, 2. , 1.5])
 
-    >>> convolve_fft([1, 0, 3], [0, 1, 0])
+    >>> convolve_fft([1, 0, 3], [0, 1, 0])  # doctest: +FLOAT_CMP
     array([ 1.00000000e+00, -3.70074342e-17,  3.00000000e+00])
 
     >>> convolve_fft([1, 2, 3], [1])

